brandonrhayes/Minimizing_Vaccination_Waste
# Input validation

All fields have been validated to ensure they are filled in and are filled in with the correct data type. Error messages show up for all errors that occur and bring the user back to the primary page instead of going to results. In the future, it would be helpful to determine how to save the user's input in the fields they inputted correctly so they don't have to start over. Copy was changed from 'doses' to 'vials' for the number of doses and dose type - easy change back if that's not agreed on, it didn't make sense to me to be saying what dose type the vial is and asking how many doses are available as that requires math from the user (ie. if they have 10 5 dose vials they have 50 doses). Further validation can occur if we need to (ie. making sure a number is between x-y, etc). 

To Test:
- Try to 'Go to results' with no input, you should have an error message for every field, plus an additional error message prompting correct input where applicable. This should total 12 pop-ups. NOTE: you will not receive an error message for not putting in the time unless you select 'yes' for terminating service
- Input anything except for numbers in the text fields, you should get pop-ups telling you the correct data type
- Select 'yes' to terminate service and try leaving the time field blank, you should get a pop-up asking to input the time and to input the time in the correct format
- Select 'yes' to terminate service and try putting text in the time field, you should get a pop-up asking to input the time in the correct format
- Select 'yes' to terminate service and try putting various numbers in the incorrect format in the time field (ie. 123345, 12, etc), you should get a pop-up asking to input the time in the correct format. NOTE: Currently it will accept 14:00, but also 1400, let me know if that's an issue.
- Select all drop downs and 'yes' for the radio button and fill in text fields correctly, you should be taken to the results page and have all data outputted to the console
- Select all drop downs and 'no' for the radio button and fill in text fields correctly, you should be taken to the results page and have all data outputted to the console
